File: backend/transactions/wards/update_ward_transaction.py
from wards.update_ward import (update_ward)
from schemas.ward_schemas import (UpdateWardRequest)
from supabase import Client

def update_ward_transaction(
                                client:Client,
                                ward: UpdateWardRequest,
                                hospital_id: str
                            ):

    print("update_ward_transaction")

    update_ward(
                    client=client, 
                    ward=ward,
                    hospital_id=hospital_id
                )

File: frontend/app/api/transactions/wards/updateWardTransaction.ts
import { API_BASE_URL } from "../../client/apiClient"
import { UpdateWardType } from "../../../types/wardTypes"
import { getWardsFromApi } from "../../wards/fetchWards"
import {  } from "../../client/apiClient"

import {
         normalizeWard,
         toUpdateWardRequest
       } from "../../../utils/wardsMapper"

type UpdateWardTransactionParams = {
                                     ward: UpdateWardType
                                     setWards: any
                                   }

export async function updateWardTransaction({
                                               ward,
                                               setWards
                                             }: UpdateWardTransactionParams
                                           )
{
  console.log("updateWardTransaction")


  await fetch(
                `${API_BASE_URL}/update-ward`,
                {
                  method: "POST",
                  headers: {
                            "Content-Type":
                            "application/json"
                            },credentials: "include",
                  body: JSON.stringify(
                                          toUpdateWardRequest(
                                                               ward
                                                             )
                                        )
                }
              )

  const wards =
    await getWardsFromApi()

  setWards(
             wards.map(
                        normalizeWard
                      )
           )
}